docs/proposals/cluster-groups.md

```diff
@@ -0,0 +1,25 @@
+# Cluster Groups
+
+## Problem
+To allow cluster-scoped group definitions to be shared across multiple related projects, we'll introduce the concept of ClusterGroups (cluster-scoped sets of users) that are managed by cluster-admins.
+
+
+## Use Cases
+ 1.  I want to be able to define some ClusterGroups that will affect policy evaluation across all projects.
+ 1.  I want to be able to efficiently manage group membership.
+
+
+## ClusterGroups
+ClusterGroups are cluster-scoped sets of users that are managed by cluster-admins and may contain any user.  ClusterGroups can be bound to roles in all projects.  That means that if UserA is added to ClusterGroupB and ProjectC has a rolebinding that says "ClusterGroupB is an editor", UserA now has editor permissions in ProjectC.  ClusterGroups will be described using an API object that looks like this:
+```
+// ClusterGroup contains the set of users that belong to a given ClusterGroup.  The name may not contain ":"
+type ClusterGroup struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	// Users is a list of Users that are members of the ClusterGroup
+	Users util.StringSet
```

Comment:
> []string for external

I had actually debated `[]LocalObjectReference` for patchability, but most of the code won't care, so I wasn't hung up on it.